cab-token-generator/javatests/com/google/auth/credentialaccessboundary/ClientSideCredentialAccessBoundaryFactoryTest.java

```diff
@@ -0,0 +1,311 @@
+              () ->
+                  ClientSideCredentialAccessBoundaryFactory.newBuilder()
+                      .setSourceCredential(sourceCredentials)
+                      .setMinimumTokenLifetime(Duration.ZERO));
+
+      assertEquals("Minimum token lifetime must be positive.", exception.getMessage());
+    }
+
+    @Test
+    public void builder_refreshMargin_negative_throws() throws IOException {
+      GoogleCredentials sourceCredentials = getServiceAccountSourceCredentials(true);
+      IllegalArgumentException exception =
+          assertThrows(
+              IllegalArgumentException.class,
+              () ->
+                  ClientSideCredentialAccessBoundaryFactory.newBuilder()
+                      .setSourceCredential(sourceCredentials)
+                      .setRefreshMargin(Duration.ofMinutes(-1)));
+
+      assertEquals("Refresh margin must be positive.", exception.getMessage());
+    }
+
+    @Test
+    public void builder_refreshMargin_zero_throws() throws IOException {
+      GoogleCredentials sourceCredentials = getServiceAccountSourceCredentials(true);
+      IllegalArgumentException exception =
+          assertThrows(
+              IllegalArgumentException.class,
+              () ->
+                  ClientSideCredentialAccessBoundaryFactory.newBuilder()
+                      .setSourceCredential(sourceCredentials)
+                      .setRefreshMargin(Duration.ZERO));
+
+      assertEquals("Refresh margin must be positive.", exception.getMessage());
+    }
+
+    @Test
+    public void builder_setsCorrectDefaultValues() throws IOException {
+      GoogleCredentials sourceCredentials = getServiceAccountSourceCredentials(true);
+      ClientSideCredentialAccessBoundaryFactory factory =
+          ClientSideCredentialAccessBoundaryFactory.newBuilder()
+              .setSourceCredential(sourceCredentials)
+              .build();
+
+      assertEquals(OAuth2Utils.HTTP_TRANSPORT_FACTORY, factory.getTransportFactory());
+      assertEquals(
+          String.format(OAuth2Utils.TOKEN_EXCHANGE_URL_FORMAT, Credentials.GOOGLE_DEFAULT_UNIVERSE),
+          factory.getTokenExchangeEndpoint());
+    }
+
+    @Test
+    public void builder_universeDomainMismatch_throws() throws IOException {
+      GoogleCredentials sourceCredentials =
+          getServiceAccountSourceCredentials(/* canRefresh= */ true);
+
+      try {
+        ClientSideCredentialAccessBoundaryFactory.newBuilder()
+            .setSourceCredential(sourceCredentials)
+            .setUniverseDomain("differentUniverseDomain")
+            .build();
+
+        fail("Should fail with universe domain mismatch.");
+      } catch (IllegalArgumentException e) {
+        assertEquals(
+            "The client side access boundary credential's universe domain must be the same as the source credential.",
+            e.getMessage());
+      }
+    }
+  }
+
+  private static GoogleCredentials getServiceAccountSourceCredentials(boolean canRefresh)
```

Comment:
nit: Can this take in a `MockTokenServerTransportFactory` param with the error/ serviceAccount configured instead of a boolean? I think that would be cleaner as this method would just be creating the Credentials instead of creating credentials and configuring the response